src/gps/nmea_parser.cc

```diff
@@ -0,0 +1,137 @@
+// BoB robotics includes
+#include "common/macros.h"
+#include "gps/nmea_parser.h"
+
+namespace BoBRobotics {
+namespace GPS {
+
+void
+parseTimeField(const std::string &field, std::tm &time)
+{
+    time.tm_hour = stoi(field.substr(0, 2));
+    time.tm_min = stoi(field.substr(2, 2));
+    time.tm_sec = stoi(field.substr(4, 2));
+
+    // Indicates that information about DST is not available
+    time.tm_isdst = -1;
+}
+
+bool
+NMEAParser::parseDateTime(const std::string &line, std::tm &time)
+{
+    if (!parse(line, "$GNZDA", 6)) {
+        return false;
+    }
+
+    try {
+        parseTimeField(m_Fields[0], time);
+        time.tm_mday = stoi(m_Fields[1]);
+        time.tm_mon = stoi(m_Fields[2]) - 1;
+        time.tm_year = stoi(m_Fields[3]) - 1900;
+
+        // This field is available in GNU/BSD but is not strict ISO C
+        int hoursOff = m_Fields[4].empty() ? 0 : stoi(m_Fields[4]);
+        int minutesOff = m_Fields[5].size() <= 3 ? 0 : stoi(m_Fields[5].substr(0, 2));
```

Comment:
what's going on with ``m_Fields[5]``? Also, what's GNU/BSD got to do with this? Aren't the fields coming straight from the GPS device? 😨

Reply:
Ok the commenting is a bit fucked here. It's actually the ``tm_gmtoff`` field of the ``std::time`` which is a GNU/BSD thing, not the fields of the NMEA sentence!

We only use the first two chars of ``m_Fields[5]`` for ``minutesOff`` because the last three chars are used for the checksum (e.g. ``*A0``)